src/kimmdy/topology/topology.py

```diff
@@ -600,14 +631,14 @@ def __init__(
             raise ValueError("molecules not found in top file")
         self.molecules = [(l[0], l[1]) for l in molecules]
 
-        self.ff = FF(top)
+        self.ff = FF(top, residuetypes_path)
         self._parse_molecules()
         self.parametrizer = parametrizer
         self._check_is_reactive_molecule = is_reactive_predicate_f
 
         self.needs_parameterization = False
 
-        self._merge_moleculetypes()
+        self._merge_moleculetypes(radicals)
```

Comment:
I think I'll have some minor cleanups about what part is responsible for keeping track of the radical list such that the moleculetype doesn't need to check again if the radical it gets are for its own use.